integration-tests/tests/admin.rs

```diff
@@ -196,3 +183,70 @@ async fn set_secret(#[case] test_config: KbsConfigType, #[case] valid_key: bool)
 
     res
 }
+
+// To avoid making the cases matrix for the other tests too complex,
+// make a separate test for the password admin interface.
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[serial]
+async fn password_admin_test() -> Result<()> {
+    let _ = env_logger::try_init_from_env(env_logger::Env::new().default_filter_or("debug"));
+
+    let mut harness =
+        TestHarness::new(KbsConfigType::EarTokenBuiltInRvpsPasswordAdmin.into()).await?;
+    harness.wait().await;
+
+    info!("TEST: logging in with invalid credentials");
+    harness
+        .login("test1".to_string(), "banana".to_string())
+        .await
+        .map_or_else(
+            |_| Ok(()),
+            |_| Err(anyhow!("Able to log in with wrong credentials.")),
+        )?;
+
+    harness
+        .login("test1".to_string(), "password2".to_string())
+        .await
+        .map_or_else(
+            |_| Ok(()),
+            |_| Err(anyhow!("Able to log in with wrong credentials.")),
+        )?;
+
+    harness
+        .login("test2".to_string(), "password1".to_string())
+        .await
+        .map_or_else(
+            |_| Ok(()),
+            |_| Err(anyhow!("Able to log in with wrong credentials.")),
+        )?;
+
+    info!("TEST: login with valid credentials");
+    harness
+        .login("test2".to_string(), "password2".to_string())
+        .await?;
+
+    info!("TEST: setting secret");
+    harness
+        .set_secret("a/b/c".to_string(), vec![0u8; 10])
+        .await?;
+
+    info!("TEST: tampering with admin token");
+    harness
+        .admin_token
+        .as_mut()
+        .unwrap()
```

Comment:
Is there any way to avoid `unwrap()` here? It's not good to me to mix `?` and `unwrap`s, as one is elagant error handling but another will panic directly